controller/engine_controller.go

```diff
@@ -679,7 +628,47 @@ func (m *EngineMonitor) stop(e *longhorn.Engine) {
 			return
 		}
 	}
-	m.stopCh <- struct{}{}
+}
+
+func (m *EngineMonitor) sync() bool {
+	for {
```

Comment:
In the case of conflict do we want to retry this forever? 
Since while we are in here, we cannot react on the close/write into the stop channel. 
- You can set the stop channel to nil after the delete above.
- Then in the for loop check whether the stop channel is nil --> return true, to signal that we require a stop.

Reply:
The above deletion is the method of `EngineController` but this sync function belongs to `EngineMonitor`. I don't think we can set the channel field to nil for the engine monitor. 
But you are right in general, there may be endless conflicts that will block the whole monitor stop. I think we can add a retry count for the retry loop.